### PR TITLE
[CI] Add release-related files and create sync group for auto-releasing repos

### PR DESCRIPTION
**Pull request by surchs**
- Closes https://github.com/neurobagel/workflows/issues/43

Changes proposed in this pull request:
- `release.yaml` and `.autorc` files used for auto-releasing added
  - `omit-commits` plugin added to exclude bot commits from changelog
- added PR template with checkbox for PR label
- sync group created for auto-releasing repos
- related template files organized into subdirectories

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [ ] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: .github/sync.yml [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
+      neurobagel/query-tool
+      neurobagel/recipes
+      neurobagel/workflows
+  
+  # Workflow and configuration files for auto-releasing with intuit/auto
+  - files:
+      - source: template_workflows/auto_release/
+        dest: .github/workflows/
+      - source: template_configs/.autorc
+        dest: .autorc
```
Comment (surchs) [inline .github/sync.yml:30]: Nice! I didn't think that we could mix files and folder destinations, but the docs seem to agree. Good find!

**Review by surchs — COMMENTED** [review]
Thanks a lot @alyssadai! I agree with your changes. Can't review "my" own PR, so could you check and approve please?

**Comment by surchs** [comment]
So for the auto-release repos we will overwrite the org-level global PR template, yes?

**Comment by alyssadai** [comment]
> So for the auto-release repos we will overwrite the org-level global PR template, yes?

yes, that's right!

**Review by alyssadai — APPROVED** [review]
@surchs, I've pushed the config for the [omit-commits](https://intuit.github.io/auto/docs/generated/omit-commits) plugin.

If it makes sense to you, this is good to merge!

**Comment by surchs** [comment]
Thanks a lot for the omit config. Looks great!

**Pull request merged**